client/lib/viewport/test/react-helpers.js

```diff
@@ -0,0 +1,218 @@
+			function TestComponent() {
+				const isActive = helpers.useBreakpoint( 'unknown' );
+				return isActive === undefined ? 'undefined' : `unexpected value: ${ isActive }`;
+			}
+
+			act( () => {
+				ReactDOM.render( <TestComponent />, container );
+			} );
+
+			expect( container.textContent ).toBe( 'undefined' );
+		} );
+
+		test( 'returns the current breakpoint state for a valid breakpoint', () => {
+			function TestComponent() {
+				const isActive = helpers.useBreakpoint( '<960px' );
+				return isActive ? 'true' : 'false';
+			}
+
+			runComponentTests( TestComponent, '(max-width: 960px)' );
+		} );
+	} );
+
+	describe( 'useMobileBreakpoint', () => {
+		test( 'returns the current breakpoint state for the mobile breakpoint', () => {
+			function TestComponent() {
+				const isActive = helpers.useMobileBreakpoint();
+				return isActive ? 'true' : 'false';
+			}
+
+			runComponentTests( TestComponent, '(max-width: 480px)' );
+		} );
+	} );
+
+	describe( 'useDesktopBreakpoint', () => {
+		test( 'returns the current breakpoint state for the desktop breakpoint', () => {
+			function TestComponent() {
+				const isActive = helpers.useDesktopBreakpoint();
+				return isActive ? 'true' : 'false';
+			}
+
+			runComponentTests( TestComponent, '(min-width: 961px)' );
+		} );
+	} );
+
+	describe( 'withBreakpoint', () => {
+		class ExpectUndefinedComponent extends React.Component {
+			render() {
+				const isActive = this.props.isBreakpointActive;
+				return isActive === undefined ? 'undefined' : `unexpected value: ${ isActive }`;
+			}
+		}
+
+		test( 'returns undefined when called with no breakpoint', () => {
+			const TestComponent = helpers.withBreakpoint( ExpectUndefinedComponent );
+
+			act( () => {
+				ReactDOM.render( <TestComponent />, container );
+			} );
+
+			expect( container.textContent ).toBe( 'undefined' );
+		} );
+
+		test( 'returns undefined for an unknown breakpoint', () => {
+			const TestComponent = helpers.withBreakpoint( ExpectUndefinedComponent, 'unknown' );
+
+			act( () => {
+				ReactDOM.render( <TestComponent />, container );
+			} );
+
+			expect( container.textContent ).toBe( 'undefined' );
+		} );
+
+		test( 'returns the current breakpoint state for a valid breakpoint', () => {
+			const TestComponent = helpers.withBreakpoint( BaseComponent, '<960px' );
+			runComponentTests( TestComponent, '(max-width: 960px)' );
+		} );
+	} );
+
+	describe( 'withMobileBreakpoint', () => {
+		test( 'returns the current breakpoint state for the mobile breakpoint', () => {
+			const TestComponent = helpers.withMobileBreakpoint( BaseComponent );
+			runComponentTests( TestComponent, '(max-width: 480px)' );
+		} );
+	} );
+
+	describe( 'withDesktopBreakpoint', () => {
+		test( 'returns the current breakpoint state for the desktop breakpoint', () => {
+			const TestComponent = helpers.withDesktopBreakpoint( BaseComponent );
+			runComponentTests( TestComponent, '(min-width: 961px)' );
+		} );
+	} );
+
+	afterAll( () => {
```

Comment:
maybe bump this up with the other lifecycle hooks at the top? I was just surprised to run into it at the end.

Reply:
Well, since it's `afterAll` it made sense in my head to be at the bottom 😄But I can see how it can be unexpected. Moved!

Reply:
😄  Maybe it's just my old C++ tendencies showing